sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -480,7 +480,7 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     try {
       Option(hive.getFunction(db, name)).map(fromHiveFunction)
     } catch {
-      case CausedBy(ex: NoSuchObjectException) if ex.getMessage.contains(name) =>
+      case CausedBy(ex: Exception) if ex.getMessage.contains(s"$name does not exist") =>
```

Comment:
Hm.. I am not familiar with this part though.. Would this be safe to catch all the exceptions?